lib/ui5Framework/AbstractResolver.js

```diff
@@ -0,0 +1,157 @@
+const path = require("path");
+const log = require("@ui5/logger").getLogger("ui5Framework:AbstractResolver");
+
+/**
+ * Abstract Resolver
+ *
+ * @public
+ * @abstract
+ * @memberof module:@ui5/project.ui5Framework
+ */
+class AbstractResolver {
+	/**
+	 * @param {*} options options
+	 * @param {string} options.version version
+	 * @param {string} [options.cwd=process.cwd()] cwd
+	 * @param {string} [options.ui5HomeDir="~/.ui5"] UI5 home dir
```

Comment:
Currently the ui5HomeDir is not resolve, so when passing a relative path, it is resolved implicitly to `process.cwd()`.
Do we want to keep this, or resolve relative paths to the passed `cwd` instead?

Let's discuss this next week.

Reply:
As discussed, the ui5HomeDir configuration might become a CLI parameter. Therefore it would make sense to have relative paths resolved relatively to `process.cwd()` (i.e. the CLI cwd).

Reply:
Maybe we should add this bit to the JSDoc here?

Reply:
Done